src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,137 @@
+    this.setState({ selectedTodo });
+  }
+
+  handleChange(name, value) {
+    this.setState({ [name]: value });
+  }
+
+  render() {
+    const {
+      todos,
+      selectUser,
+    } = this.props;
+
+    const {
+      selectedTodo,
+      title,
+      select,
+    } = this.state;
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
 
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
+        <div className="TodoList__list-container">
+          <label htmlFor="title-input">
+            Search by title:
+            <input
+              className="TodoList__title-input"
+              placeholder="filter todos"
+              name="title"
+              id="title-input"
+              type="text"
+              value={title}
+              onChange={(event) => {
+                const { name, value } = event.target;
+
+                this.handleChange(name, value);
+              }}
+            />
+            <select
+              className="TodoList__select"
+              name="select"
+              value={select}
+              onChange={(event) => {
+                const { name, value } = event.target;
+
+                this.handleChange(name, value);
+              }}
+            >
+              <option>all</option>
+              <option>active</option>
+              <option>completed</option>
+            </select>
           </label>
+          <ul className="TodoList__list">
+            {todos.filter((todo) => {
+              if (select === 'all') {
+                return true;
+              }
+
+              if (select === 'active') {
+                return !todo.completed;
+              }
+
+              return todo.completed;
+            }).map(todo => (
+              todo.title.includes(title) && (
+              <li
+                key={todo.id}
+                className={classNames('TodoList__item', {
+                  'TodoList__item--checked': todo.completed,
+                  'TodoList__item--unchecked': !todo.completed,
+                })}
+              >
+                <label>
+                  <input
+                    className={classNames('input')}
+                    type="checkbox"
+                    checked={todo.completed}
+                  />
+                  <p>
+                    {todo.title}
+                  </p>
+                </label>
+
+                <button
+                  className={classNames('button', {
+                    'TodoList__user-button--selected': todo.id === selectedTodo,
```

Comment:
у тебя тут неправильно и из-за этого неправильно работает подсвечивание кнопок, когда нажимаешь на юзера должны подсвечиваться все кнопки с этим юзером, а у тебя только одна подсвечивается 
```suggestion
                    'TodoList__user-button--selected': todo.userId === selectedTodo,
```

Reply:
в задании не было сказано что должны подсвечиваться все кнопки с юзером...

Reply:
как скажешь =)

Reply:
и наверное более логично будет если конкретная кнопка у конкретной тудушки будет подсвечиваться при нажатии на нее, а не все ....